soak/README.md

```diff
@@ -0,0 +1,274 @@
+## Introduction
+
+ACK team provides the soak testing framework where service teams can execute long running soak tests for testing the
+performance and stability of the service controllers. Running soak tests is also a requirement for cutting the stable
+release for the service controller. Find more details on the release process [here](https://github.com/aws-controllers-k8s/community/blob/main/docs/contents/releases.md).
+
+Under the hood, the soak test introduces load by repeatedly running the end-to-end tests for the service controller. End
+to end tests are present in service-controller github repository. i.e. "https://github.com/aws-controllers-k8s/<service-name>-controller".
+
+If you wish to run soak tests for your service controller locally, see the "HOW TO RUN SOAK TEST LOCALLY" section below.
+
+## Running Soak Tests Using ACK Prow Jobs
+
+TBD
+
+
+## How To Run Soak Test Locally
+
+TODO
+```
+* Convert following 9 steps into a single executable script with default ACK soak-test params. 
+* Service teams will have the freedom to set parameters and customizing their soak-test runs.
+```
+
+### Prerequisites
+* A tool for building OCI images (docker, buildah etc..)
+* A Kubernetes cluster to run soak tests
+* helm 
+* kubectl
+* curl
+
+### Step 0 (Declare the ACK soak test configuration)
+
+* Setup soak test directory
+```bash
+mkdir -p ~/.ack/soak && cd ~/.ack/soak
+```
+
+* Create a file name `config` and copy the following configuration into that file, substituting "< placeholders >" with
+actual values.
+
+```
+# NOTE: Substitute all the <placeholders> with actual values.
+# A handy alias to go to root of ACK soak test directory
+alias go-to-soak='cd ~/.ack/soak'
+
+# SERVICE CONTROLLER CONFIGURATION
+# AWS Service name of the controller under test. Ex: apigatewayv2
+export SERVICE_NAME=<aws-service-name>
+# Semver version of the controller under test. Ex: v0.0.2
+export CONTROLLER_VERSION=<semver>
+export CONTROLLER_CHART_URL=public.ecr.aws/aws-controllers-k8s/chart:$SERVICE_NAME-$CONTROLLER_VERSION
+# Release name of controller helm chart. Ex: soak-test
+export CONTROLLER_CHART_RELEASE_NAME=<controller-chart-release-name>
+
+# PROMETHEUS, GRAFANA CONFIGURATOIN
+# Release name of kube-prometheus helm chart. Ex: prom
+export PROM_CHART_RELEASE_NAME=<kube-prometheus-chart-release-name>
+# Local port to access Prometheus dashbaord
+export LOCAL_PROMETHEUS_PORT=9090
+# Local port to access Prometheus dashbaord
+export LOCAL_GRAFANA_PORT=3000
+
+# SOAK TEST RUNNER CONFIGURATION
+# Git branch of service controller repository where e2e tests are present.
+export E2E_GIT_REF=main
```

Comment:
This should be the git commit of the tag (`CONTROLLER_VERSION`)? Otherwise tests may be out of sync with the release version